src/main/java/com/google/cloud/tools/maven/deploy/AbstractDeployMojo.java

```diff
@@ -14,124 +14,99 @@
  */
 
-package com.google.cloud.tools.maven;
+package com.google.cloud.tools.maven.deploy;
 
-import com.google.cloud.tools.appengine.api.deploy.DeployConfiguration;
-import com.google.cloud.tools.appengine.api.deploy.DeployProjectConfigurationConfiguration;
-import java.io.File;
-import java.util.ArrayList;
-import java.util.List;
+import com.google.cloud.tools.maven.stage.AbstractStageMojo;
 import org.apache.maven.plugins.annotations.Parameter;
 
-public abstract class AbstractDeployMojo extends AbstractStageMojo
-    implements DeployConfiguration, DeployProjectConfigurationConfiguration {
-
-  /** This is not configurable by the user. */
-  private List<File> deployables = new ArrayList<>();
+/** Mojo configuration for Deploy with Staging inherited */
+public abstract class AbstractDeployMojo extends AbstractStageMojo {
 
   /**
    * The Google Cloud Storage bucket used to stage files associated with the deployment. If this
    * argument is not specified, the application's default code bucket is used.
    */
   @Parameter(alias = "deploy.bucket", property = "app.deploy.bucket")
-  private String bucket;
+  protected String bucket;
 
   /**
    * Deploy with a specific Docker image. Docker url must be from one of the valid gcr hostnames.
    *
    * <p><i>Supported only for app.yaml based deployments.</i>
    */
   @Parameter(alias = "deploy.imageUrl", property = "app.deploy.imageUrl")
-  private String imageUrl;
+  protected String imageUrl;
 
   /** Promote the deployed version to receive all traffic. True by default. */
   @Parameter(alias = "deploy.promote", property = "app.deploy.promote")
-  private Boolean promote;
+  protected Boolean promote;
 
   /** The App Engine server to connect to. You will not typically need to change this value. */
   @Parameter(alias = "deploy.server", property = "app.deploy.server")
-  private String server;
+  protected String server;
 
   /** Stop the previously running version when deploying a new version that receives all traffic. */
   @Parameter(alias = "deploy.stopPreviousVersion", property = "app.deploy.stopPreviousVersion")
-  private Boolean stopPreviousVersion;
+  protected Boolean stopPreviousVersion;
 
   /**
    * The version of the app that will be created or replaced by this deployment. If you do not
    * specify a version, one will be generated for you.
    */
   @Parameter(alias = "deploy.version", property = "app.deploy.version")
-  private String version;
+  protected String version;
 
   /** The Google Cloud Platform project Id to use for this invocation */
   @Deprecated
   @Parameter(alias = "deploy.project", property = "app.deploy.project")
-  private String project;
+  protected String project;
 
   /** The Google Cloud Platform project Id to use for this invocation. */
   @Parameter(alias = "deploy.projectId", property = "app.deploy.projectId")
-  private String projectId;
+  protected String projectId;
+
+  protected Deployer.Factory deployerFactory = new Deployer.Factory();
```

Comment:
Can this be `final`? If so, why does this have to be defined as a field?

Reply:
Without a constructor though, they need to be non final to mock. Maybe I should add the constructor

Reply:
Ah, I see declaring it as a field is to inject a mock instance for testing.